views/RepoList.tsx

```diff
@@ -14,9 +14,9 @@ const RepoList = () => {
 	useEffect(() => {
 		setLoading(true);
 		// get the list of repositories of the user
-		axios.get<{repoList: DbRepoSerializable[]}>("/api/repoList")
+		axios.get<{ repoList: DbRepoSerializable[] }>(`/api/repoList?nonce=${Math.random()}`) // adding query parameter to bypass cache
```

Comment:
So apparently, this is a better way to bypass cache compared to the `'Cache-Control': 'no-cache'` header in the API call.
Here are the reasons, as provided by Claude 3 Sonnet model:

> Both methods are valid and can be used to prevent caching for the /api/repoList request in the RepoList component. However, the recommended approach is to use the nanoid library to generate a unique query parameter for each request.
>
>Using a unique query parameter is generally considered a better approach for the following reasons:
>
>**Simplicity**: Adding a query parameter is a straightforward and widely used technique for preventing caching. It doesn't require any additional headers or configuration on the server-side.
>
>**Flexibility**: By using a unique query parameter, you can selectively disable caching for specific requests without affecting the caching behavior of other requests or resources.
>
>**Performance**: Disabling caching entirely by setting headers like `Cache-Control: no-cache` can have a negative impact on performance, as it prevents the browser from caching any resources for that request. Using a unique query parameter allows the browser to cache other resources (like static assets) while preventing caching for the specific API endpoint.
>
>**Compatibility**: Some older browsers or proxy servers may not respect the `Cache-Control` header correctly, leading to inconsistent caching behavior. Using a unique query parameter is a more reliable way to prevent caching across different browsers and environments.
>
>**Separation of Concerns**: By using a unique query parameter, you can handle caching concerns at the client-side without modifying the server-side code or headers.
>
>That being said, both approaches are valid and widely used. If you have specific requirements or constraints in your application, you may choose the approach that best fits your needs.